### PR TITLE
Add CNB (Czech National Bank) provider

### LINKED ISSUE #280: Add CNB (Czech National Bank) provider
## Czech National Bank

- **Key:** CNB
- **Base currency:** CZK
- **Currencies:** 56 vs CZK
- **API:** REST with Swagger docs at https://api.cnb.cz/cnbapi/swagger-ui.html
- **Format:** JSON, CSV
- **Publish schedule:** Daily (weekdays)

### Notes

Well-documented Swagger API, no auth required. 56 currencies is strong coverage. CZK adds a Central European base not covered by other providers.

### Checklist

See [.claude/skills/add-provider.md](../blob/main/.claude/skills/add-provider.md)

### PR DESCRIPTION
**Pull request by lineoffligbot**
## Summary

- Add Czech National Bank as a new exchange rate data provider
- Daily rates for 30 currencies against CZK via the CNB REST API
- Per-day fetch with amount normalization (HUF/100, IDR/1000, etc.) and chunked backfill
- Historical data back to 1991

Closes #280

## Test plan

- [x] Integration test with live VCR cassette
- [x] Parse unit tests for amount normalization (1, 100, 1000)
- [x] Zero-rate edge case
- [x] Full test suite passes (390 tests, 0 failures)
- [x] RuboCop clean
- [ ] `bundle exec rake db:seed` registers provider
- [ ] `bundle exec rake backfill[cnb]` backfills successfully

### CODE REVIEW

File: spec/providers/cnb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    let(:provider) { CNB.new }
+
+    def count_unique_dates
+      Rate.select(:date).distinct.count
+    end
+
+    it "fetches rates with date range" do
+      provider.fetch(since: Date.new(2026, 3, 16), upto: Date.new(2026, 3, 20)).import
+
+      _(count_unique_dates).must_be(:>=, 1)
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply spec/providers/cnb_spec.rb]: Tightened to >= 3. Hardcoding 5 is fragile if the range spans a holiday.

File: lib/providers/cnb.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+      rates.filter_map do |r|
+        code = r["currencyCode"]
+        next unless code&.match?(/\A[A-Z]{3}\z/)
+
+        amount = r["amount"].to_f
+        rate = r["rate"].to_f
+        next if rate.zero? || amount.zero?
+
+        date = Date.parse(r["validFor"])
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply lib/providers/cnb.rb:51]: Added per-row rescue in d1029ce.

File: spec/vcr_cassettes/cnb.yml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        Africa","currency":"rand","amount":1,"currencyCode":"ZAR","rate":1.270},{"validFor":"2026-03-16","order":52,"country":"South
+        Korea","currency":"won","amount":100,"currencyCode":"KRW","rate":1.428},{"validFor":"2026-03-16","order":52,"country":"Sweden","currency":"krona","amount":1,"currencyCode":"SEK","rate":2.269},{"validFor":"2026-03-16","order":52,"country":"Switzerland","currency":"franc","amount":1,"currencyCode":"CHF","rate":27.022},{"validFor":"2026-03-16","order":52,"country":"Thailand","currency":"baht","amount":100,"currencyCode":"THB","rate":65.805},{"validFor":"2026-03-16","order":52,"country":"Turkey","currency":"lira","amount":100,"currencyCode":"TRY","rate":48.169},{"validFor":"2026-03-16","order":52,"country":"United
+        Kingdom","currency":"pound","amount":1,"currencyCode":"GBP","rate":28.280},{"validFor":"2026-03-16","order":52,"country":"USA","currency":"dollar","amount":1,"currencyCode":"USD","rate":21.284}]}'
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply spec/vcr_cassettes/cnb.yml]: This is default VCR serialization. Not worth hand-editing.
